rotkehlchen/exchanges/binance.py

```diff
@@ -1075,3 +1077,13 @@ def query_online_margin_history(
             end_ts: Timestamp,  # pylint: disable=unused-argument
     ) -> List[MarginPosition]:
         return []  # noop for binance
+
+    def get_selected_pairs(self) -> List[BinancePair]:
+        pairs = self.db.get_binance_pairs(self.name, self.location)
+        validated_pairs = []
+        if pairs:
+            for pair in pairs:
+                known_pair = self._symbols_to_pair.get(pair)
+                if known_pair is not None:
+                    validated_pairs.append(known_pair)
+        return validated_pairs
```

Comment:
This is too complicated. If you want to follow this approach make a set of the `_symbols_to_pair` and a set of the `pairs` and then get their union. Faster and easier to read.

But ... shouldn't you actually verify that a pair is valid **before** saving it in the database?

Which I don't see you doing at all.

On the other hand even if you verify it they may delist the pair in the meantime, but I would like to verify any data going in the DB.

Reply:
The problem of verifying this at the moment of inserting is that it implies a new query to the binance api and then I would need to make the request async in the frontend what is a bit more complicated.